homeassistant/components/intellifire/config_flow.py

```diff
@@ -71,31 +78,101 @@ async def _find_fireplaces(self):
         LOGGER.debug("Configured Hosts: %s", configured_hosts)
         LOGGER.debug("Not Configured Hosts: %s", self._not_configured_hosts)
 
-    async def _async_validate_and_create_entry(self, host: str) -> FlowResult:
-        """Validate and create the entry."""
+    async def validate_api_access_and_create_or_update(
+        self, *, host: str, username: str, password: str, serial: str
+    ):
+        """Validate username/password against api."""
+        ift_control = IntellifireControlAsync(fireplace_ip=host)
+
+        LOGGER.info("Attempting login with: %s %s", username, password)
+        # This can throw an error which will be handled above
+        try:
+            await ift_control.login(username=username, password=password)
+            await ift_control.get_username()
+        finally:
+            await ift_control.close()
+
+        data = {CONF_HOST: host, CONF_PASSWORD: password, CONF_USERNAME: username}
+
+        # Update or Create
+        existing_entry = await self.async_set_unique_id(DOMAIN)
```

Comment:
Why set `DOMAIN` as unique ID, that means only one can exist ? You have a serial and that should be the ID ?